src/config.rs

```diff
@@ -401,110 +402,95 @@ impl ProjectPathsConfig {
-        self.flatten_node(target, &graph, &mut Default::default(), false, false, false).map(|x| {
-            format!("{}\n", utils::RE_THREE_OR_MORE_NEWLINES.replace_all(&x, "\n\n").trim())
-        })
-    }
-
-    /// Flattens a single node from the dependency graph
-    fn flatten_node(
-        &self,
-        target: &Path,
-        graph: &Graph,
-        imported: &mut HashSet<usize>,
-        strip_version_pragma: bool,
-        strip_experimental_pragma: bool,
-        strip_license: bool,
-    ) -> Result<String> {
-        let target_dir = target.parent().ok_or_else(|| {
-            SolcError::msg(format!("failed to get parent directory for \"{:?}\"", target.display()))
-        })?;
-        let target_index = graph.files().get(target).ok_or_else(|| {
-            SolcError::msg(format!("cannot resolve file at {:?}", target.display()))
-        })?;
-
-        if imported.contains(target_index) {
-            // short circuit nodes that were already imported, if both A.sol and B.sol import C.sol
-            return Ok(String::new());
-        }
-        imported.insert(*target_index);
-
-        let target_node = graph.node(*target_index);
-
-        let mut imports = target_node.imports().clone();
-        imports.sort_by_key(|x| x.loc().start);
+        let ordered_deps = collect_ordered_deps(&flatten_target, self, &graph)?;
+
+        let mut sources = Vec::new();
+
+        let mut result = String::new();
+
+        for path in ordered_deps.iter() {
+            let node_id = graph.files().get(path).ok_or_else(|| {
+                SolcError::msg(format!("cannot resolve file at {}", path.display()))
+            })?;
+            let node = graph.node(*node_id);
+            let mut content = node.content().to_owned();
+
+            for alias in node.imports().iter().flat_map(|i| i.data().aliases()) {
+                let (alias, target) = match alias {
+                    SolImportAlias::Contract(alias, target) => (alias.clone(), target.clone()),
+                    _ => continue,
+                };
+                let name_regex = utils::create_contract_or_lib_name_regex(&alias);
+                let target_len = target.len() as isize;
+                let mut replace_offset = 0;
+                for cap in name_regex.captures_iter(&content.clone()) {
+                    if cap.name("ignore").is_some() {
+                        continue;
+                    }
+                    if let Some(name_match) =
+                        ["n1", "n2", "n3"].iter().find_map(|name| cap.name(name))
+                    {
+                        let name_match_range =
+                            utils::range_by_offset(&name_match.range(), replace_offset);
+                        replace_offset += target_len - (name_match_range.len() as isize);
+                        content.replace_range(name_match_range, &target);
+                    }
+                }
+            }
```

Comment:
could we add a bit of docs over what's happening here?

Reply:
will add, however this code was written before me so I might get something wrong haha